SonarQube.TeamBuild.PostProcessor/SummaryReportBuilder.cs

```diff
@@ -107,18 +107,18 @@ private void GenerateReports()
                 throw new ArgumentNullException("result");
             }
 
+            var validProjects = result.GetProjectsByStatus(ProjectInfoValidity.Valid);
 
-            SummaryReportData summaryData = new SummaryReportData
+            var summaryData = new SummaryReportData
             {
-                SkippedProjects = GetProjectsByStatus(result, ProjectInfoValidity.NoFilesToAnalyze).Count(),
-                InvalidProjects = GetProjectsByStatus(result, ProjectInfoValidity.InvalidGuid).Count()
+                SkippedProjects = result.GetProjectsByStatus(ProjectInfoValidity.NoFilesToAnalyze).Count,
+                InvalidProjects = result.GetProjectsByStatus(ProjectInfoValidity.InvalidGuid).Count,
+                ExcludedProjects = result.GetProjectsByStatus(ProjectInfoValidity.ExcludeFlagSet).Count,
+                ProductProjects = validProjects.Count(p => p.ProjectType == ProjectType.Product),
+                TestProjects = validProjects.Count(p => p.ProjectType == ProjectType.Test),
             };
-            summaryData.InvalidProjects += GetProjectsByStatus(result, ProjectInfoValidity.DuplicateGuid).Count();
 
-            summaryData.ExcludedProjects = GetProjectsByStatus(result, ProjectInfoValidity.ExcludeFlagSet).Count();
-            IEnumerable<ProjectInfo> validProjects = GetProjectsByStatus(result, ProjectInfoValidity.Valid);
-            summaryData.ProductProjects = validProjects.Count(p => p.ProjectType == ProjectType.Product);
-            summaryData.TestProjects = validProjects.Count(p => p.ProjectType == ProjectType.Test);
+            summaryData.InvalidProjects += result.GetProjectsByStatus(ProjectInfoValidity.DuplicateGuid).Count;
```

Comment:
We are calling a lot the `GetProjectsByStatus` method with different filters. Don't we have a way to get the collection and to do Linq filtering on it? It would probably be better (performance) and would definitely make the `InvalidProjects` easier to read.